### PR TITLE
Add User drop up to sidebar

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Add User drop up to sidebar

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Review user panel in sidebar

### CODE REVIEW

File: src/components/User/User.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from "react";
```
Comment (hatched) [inline src/components/User/User.js]: `User` seems like a pretty generic name. Maybe `UserMenu`? This also might be a problem with the `user` class below being pretty generic which would be solved by going to `user-menu`.

File: src/components/User/User.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  return (
+    <div
+      className={classNames("user", {
+        "is-expanded": isExpanded
+      })}
+    >
+      <div className="user__header" onClick={() => setIsExpanded(!isExpanded)}>
+        <i className="p-icon--user"></i>
+        {/* @TODO Add user name here */}
```
Comment (hatched) [inline src/components/User/User.js]: Is there a follow-up issue for this?

**Comment by hatched** [comment]
QA Notes:

When you open the menu and click 'settings' it snaps closed. It might be nice if it either stayed open on those types of panes, or animated back closed. When you click the `settings` link if you're already in the settings view it stays open.

File: src/components/User/User.js (hunk 2 above)
Comment (barrymcgee) [reply src/components/User/User.js]: https://github.com/canonical-web-and-design/jaas-dashboard/issues/365

File: src/components/UserMenu/UserMenu.test.js [hunk 3]
```diff
@@ -14,30 +14,30 @@ describe("User Icon", () => {
     const wrapper = mount(
       <Provider store={store}>
         <Router>
-          <UserIcon />
+          <UserMenu />
         </Router>
       </Provider>
     );
-    expect(wrapper.find(".user-icon")).toMatchSnapshot();
+    expect(wrapper.find(".user")).toMatchSnapshot();
```
Comment (hatched) [inline src/components/UserMenu/UserMenu.test.js]: I think you missed an update here.

**Comment by barrymcgee** [comment]
@hatched Can you add your design enhancement as a separate ticket please?

**Comment by hatched** [comment]
Sure thing

**Comment by hatched** [comment]
https://github.com/canonical-web-and-design/jaas-dashboard/issues/366

**Pull request merged**